src/semgrep_mcp/server.py

```diff
@@ -400,13 +274,32 @@ def remove_temp_dir_from_results(results: SemgrepScanResult, temp_dir: str) -> N
 # MCP Server
 # ---------------------------------------------------------------------------------
 
+
+@asynccontextmanager
+async def server_lifespan(_server: Server) -> AsyncIterator[SemgrepContext]:
+    """Manage server startup and shutdown lifecycle."""
+    # Initialize resources on startup
+    # MCP requires Pro Engine
+    process = await run_semgrep_process(["mcp", "--pro"])
```

Comment:
I'm not too familiar with the semgrep cli, what does this command run? Does it start an rpc server for the cli?

Reply:
This starts the new `semgrep mcp` process, which is a JSONRPC loop that can be sent commands to execute. In particular, this process will be able to do things like cache rules, run Semgrep analysis on the file system, etc. This should make us much more performant, and unlock us from being restricted to the Semgrep CLI.